collection+xattrs.go

```diff
@@ -618,16 +567,63 @@ func removeUserXattrs(xattrs semiParsedXattrs) {
 	}
 }
 
-// Sets JSON properties "cas" to the given `cas`, and "value_crc" to CRC checksum of `docValue`.
-func (e *event) expandSyncXattrMacros(xattr any) {
-	if xattrMap, ok := xattr.(map[string]any); ok {
-		// For some reason Server encodes `cas` as 8 hex bytes in little-endian order...
-		casBytes := make([]byte, 8)
-		binary.LittleEndian.PutUint64(casBytes, e.cas)
-		xattrMap["cas"] = fmt.Sprintf("0x%x", casBytes)
+// macroExpand returns the string representation for the specified expansion for the event
+func (e *event) macroExpand(expansion sgbucket.MacroExpansionType) (string, error) {
+
+	switch expansion {
+	case sgbucket.MacroCas:
+		return e.casAsString(), nil
+	case sgbucket.MacroCrc32c:
+		return encodedCRC32c(e.value), nil
+	default:
+		return "", fmt.Errorf("Unsupported MacroExpansionType: %v", expansion)
+	}
+}
+
+// casAsString generates the string representation of CAS used by Couchbase Server's macro expansion
+func casAsString(value CAS) string {
```

Comment:
nit: should this use `fmt.Stringer` ?

Reply:
Intentionally avoided this as I think we'd prefer to report CAS normally when logging, and only use this format for compatibility with CBS.